src/modules/bspwm.cpp

```diff
@@ -204,10 +208,27 @@ namespace modules {
       if ((pos = data.find(BSPWM_STATUS_PREFIX)) != string::npos) {
         data = data.replace(pos, prefix_len, ":");
       }
+
+      // workaround to get monitor index
+      {
+        auto f = [&positions, &data](const std::string& sub) {
+          size_t pos = data.find(sub, 0);
+          while (pos != string::npos) {
+            positions.insert(pos);
+            pos = data.find(sub, pos + 1);
+          }
+        };
+
+        f(":m");
+        f(":M");
+      }
+
       if ((pos = data.find(needle_active)) != string::npos) {
+        curr_pos = pos;
         data.erase(0, pos + 1);
       }
       if ((pos = data.find(needle_inactive)) != string::npos) {
+        curr_pos = pos;
```

Comment:
Wouldn't it be easier to just count the number of occurences of `:M` and `:m` before `needle_active` or `needle_inactive`?

Something like this (pseudocode):

```
found_pos = data.find(needle_active))

if (found_pos == string::npos) {
  found_pos = data.find(needle_inactive)
}

{
  auto count = (string, end_pos, needle) {
     // Counts the number of occurences of `needle` in string[0, end_pos)
  }
  index = count(data, found_pos, ":m") + count(data, found_pos, ":M") + 1;
}

data.erase(0, found_pos + 1);
```

Then we could avoid the kind of dirty hack using the ordered set.